devbox.json

```diff
@@ -0,0 +1,20 @@
+{
+  "packages": [
+    "bash@latest",
+    "clusterctl@latest",
+    "envsubst@latest",
+    "gnumake@latest",
+    "ginkgo@2.1.4",
+    "go@1.21.5",
+    "gotestsum@1.6.4",
+    "kind@0.17.0",
```

Comment:
while we are at at, lets upgrade kind to 0.20.0

Reply:
i dont know about the cgroups changes and if they will work in the CI environment

Reply:
no, let's not do that here.

Reply:
just documenting the reason: per https://endoflife.date/kubernetes 1.25 is eol and with kind 0.17.0 i see we are creating k8s with 1.25.3 <pre>
 devbox-build  make kind-create
kind create cluster --name=capi-test
Creating cluster "capi-test" ...
 ✓ Ensuring node image (kindest/node:v1.25.3) 🖼 
 ✓ Preparing nodes 📦  
 ✓ Writing configuration 📜 
 ✓ Starting control-plane 🕹️ 
 ✓ Installing CNI 🔌 
 ✓ Installing StorageClass 💾 
Set kubectl context to "kind-capi-test"
You can now use your cluster with:

kubectl cluster-info --context kind-capi-test
</pre>
please file a tracker if we intend to do in separate PR

Reply:
well aware of the reason but we need some changes on our side before we can accommodate a kind upgrade. There's no point blocking this PR / changing the scope of this PR due to those changes (see older comments on the PR: https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/358#discussion_r1449426652 and https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/358#discussion_r1449430470)

Reply:
we already have trackers for getting those issues fixed.